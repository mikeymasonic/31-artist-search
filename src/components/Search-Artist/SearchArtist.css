@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@200;300&display=swap');

.searchArtist {
  display: flex;
  justify-content: center;
  align-items: center;
}

.searchArtist button {
  background-color: rgb(133, 220, 226);
  color: white;
  width: 8vw;
  height: 2.6vw;
  font-size: 1vw;
  text-align: center;
  padding: 0;
  font-family: 'Montserrat', sans-serif;
}

.searchArtist button:hover {
  background-color: hotpink;
}

.searchArtist input {
  width: 14vw;
  height: 2vw;
  font-size: 1.4vw;
  font-family: 'Montserrat', sans-serif;
  margin-right: .5vw;
  border: dotted .1vw rgb(197, 197, 197);
  color: rgb(93, 93, 93);
}