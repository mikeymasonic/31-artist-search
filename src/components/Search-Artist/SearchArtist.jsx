import React from 'react';
import PropTypes from 'prop-types';
import styles from './SearchArtist.css';

const SearchArtist = ({ artist, onInputChange, onButtonClick }) => {
  return (
    <form onSubmit={onButtonClick} className={styles.searchArtist}>
      <input type='text' name='artist' value={artist} onChange={onInputChange} />
      <button type='submit'>search</button>
    </form>
  );
};

SearchArtist.propTypes = {
  artist: PropTypes.string.isRequired,
  onInputChange: PropTypes.func.isRequired,
  onButtonClick: PropTypes.func.isRequired
};

export default SearchArtist;
