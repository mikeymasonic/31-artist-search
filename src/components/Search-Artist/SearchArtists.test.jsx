import React from 'react';
import { shallow } from 'enzyme';
import SearchArtist from './SearchArtist';

describe('SearchArtist component', () => {
  it('renders SearchArtist', () => {
    const wrapper = shallow(<SearchArtist artist='Silkworm' onButtonClick={() => {}} onInputChange={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
