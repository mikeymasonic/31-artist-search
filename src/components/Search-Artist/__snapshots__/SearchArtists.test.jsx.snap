// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchArtist component renders SearchArtist 1`] = `
<form
  className="searchArtist"
  onSubmit={[Function]}
>
  <input
    name="artist"
    onChange={[Function]}
    type="text"
    value="Silkworm"
  />
  <button
    type="submit"
  >
    search
  </button>
</form>
`;
